### DISCUSSION TITLE
Most idiomatic Civet for indexing an array by an array of indices?

### DISCUSSION THREAD

**Discussion by gwhitney** [DanielXMoore/Civet#951] [Q&A]
Suppose I have an array `fruit := ['apples', 'oranges', 'pears', 'bananas', 'mangoes']` and an array `IX := [1,3,0,3]` and I'd like to index `fruit` by each element of `IX` to get `['oranges', 'bananas', 'apples', 'bananas']`. What's the nicest/most Civetitious way to do this?

I've thought of `IX.map ($) => fruit[$]` but there it's sort of annoying that I can't just do `IX.map fruit[&]` (at least not unless there's some motion on #480). Also, this seems to put the emphasis on IX when conceptually fruit is more the "subject of this sentence." I'd rather write something more along the lines of `fruit[IX]` although of course I know that doesn't work. There isn't any notion for this sort of "random-access slice," is there? From the perspective of emphasizing the fruit, `[0...IX.length].map ($) => fruit[IX[$]]` is kind of better, but it's sort of verbose/klunky, too, and I imagine a bit less efficient if I am doing a big case of this.

So at the moment I am just doing:
```
by := (arr, ixes) => ixes.map ($) => arr[$]
operator by 
```
and then writing `fruit by IX`. Have I missed anything nicer?

**Comment by bbrk24** [comment]
> Also, this seems to put the emphasis on IX when conceptually fruit is more the "subject of this sentence."

Isn't that what the pipe operator is for?

```civet
((i) -> fruit[i]) |> IX.map
```

**Comment by gwhitney** [comment]
That's not a bad suggestion, thanks! But after noodling around a bit, I think I like `fruit[ix] for each ix of IX` a bit better. Although it uses a lot of words, it more comfortably for me captures what I feel like I am doing conceptually.
